tax-processing-functions/quality_and_split/main.py

```diff
@@ -0,0 +1,403 @@
+                    table_name,
+                    extracted_1040_doc_processed,
+                    file_name,
+                    "1040sc",
+                )
+                print("1040sc output has been written to BQ.")
+
+                # Publish to pubsub
+                topic_path = publisher.topic_path(project_id, topic_name)
+                publish_to_pubsub(topic_path, json_string)
+
+            elif "1120s" in str(entity.type):
+                # move to 1040c input bucket
+                move_blob(input_bucket_name, bucket_1120s_input, blob_name=file_name)
+
+                # extract and format entities from document
+                extracted_1120_doc = extract_entities(
+                    file, content_type, processor_1120
+                )  # parse 1120s doc
+                extracted_1120_doc_processed = format_entities(
+                    extracted_1120_doc
+                )  # formatted
+                json_string = json.dumps(extracted_1120_doc_processed, sort_keys=False)
+
+                # Get timestamp and upload to GCS
+                stamp = datetime.now()
+                str_date_time = stamp.strftime(ts_format)
+                upload_blob(
+                    bucket_1120s_processed,
+                    f"1120sk_{str_date_time}_{file_name}",
+                    json_string,
+                )
+
+                # Write formatted data to BQ Table
+                write_to_bq(
+                    project_name,
+                    dataset_name,
+                    table_name,
+                    extracted_1120_doc_processed,
+                    file_name,
+                    "1120ssk1",
+                )
+                print("1120s output has been written to BQ.")
+
+                # Publish to pubsub
+                topic_path = publisher.topic_path(project_id, topic_name)
+                publish_to_pubsub(topic_path, json_string)
+
+            # TODO: Add handling for type other if needed
+    return
+
+
+# Format extracted JSON output to a list
+def format_entities(extracted_doc):
+    # extracted_doc = results from DocAI processor
+    result_ents = []
+    for entity in extracted_doc.entities:
+        entity_type = str(entity.type_)
+        if entity_type == "line_item":
+            for p in entity.properties:
+                ents = {
+                    "type_": p.type_,
+                    "mention_text": p.mention_text,
+                    "confidence": p.confidence,
+                }
+                result_ents.append(ents)
+            ents = {
+                "type_": entity.type_,
+                "mention_text": entity.mention_text,
+                "confidence": entity.confidence,
+            }
+            result_ents.append(ents)
+        else:
+            ents = {
+                "type_": entity_type,
+                "mention_text": entity.mention_text,
+                "confidence": entity.confidence,
+            }
+            result_ents.append(ents)
```

Comment:
```suggestion
    for entity in extracted_doc.entities:
        entity_type = str(entity.type_)
        for p in entity.properties:
            ents = {
                "type_": p.type_,
                "mention_text": p.mention_text,
                "confidence": p.confidence,
            }

        ents = {
            "type_": entity_type,
            "mention_text": entity.mention_text,
            "confidence": entity.confidence,
        }
        result_ents.append(ents)
```

Theres more than one entity type that can have `properties` or "nested entities" I'd recommend doing something like this to avoid duplication